python/lsst/pipe/tasks/makeCoaddTempExp.py

```diff
@@ -745,9 +745,11 @@ def runQuantum(self, butlerQC, inputRefs, outputRefs):
         PipelineTask (Gen3) entry point to warp and optionally PSF-match
         calexps. This method is analogous to `runDataRef`.
         """
-
-        # Ensure all input lists are in same detector order as the calExpList
+        # Obtain the list of input detectors from calExpList.  Sort them by
+        # detector order (to ensure reproducibility).  Then ensure all input
+        # lists are in the same sorted detector order.
         detectorOrder = [ref.datasetRef.dataId['detector'] for ref in inputRefs.calExpList]
+        detectorOrder.sort()
```

Comment:
Is this because gen2's `groupPatchExposures` sorts by detectorId?

Reply:
Ah, good point.  I'm not actually sure if there is a forced ordering in `groupPatchExposures`, but it at least happens naturally the way we (I) currently run Gen2 (but could possibly be made to have a different ordering by specifically arranging them in the `--selectId`, but I'm not worrying about this as the only one running Gen2 for these types of comparisons is yours truly!) With Gen3, this ordering is currently not guarantee-able (but there is DM-31701 to enable this).  Sorting here forces the matter.